src/Missings.jl

```diff
@@ -165,4 +165,24 @@ function levels(x)
     levs
 end
 
+struct PassMissing{F} <: Function end
+
+(::PassMissing{F})(xs...;kw...) where {F} =
+     any(ismissing, xs) || any(ismissing, values(values(kw))) ? missing : F(xs...; kw...)
+
+"""
+    passmissing(f)
+
+Return a function that returns `missing` if any of its positional or keyword arguments
+are `missing` and otherwise applies `f` to those arguments.
+
+# Examples
+```jldoctest
+julia> passmissing(sqrt).([missing, 4])
+2-element Array{Union{Missing, Float64},1}:
+  missing
+   2.0
+"""
+passmissing(f::Base.Callable) = PassMissing{f}()
```

Comment:
Shouldn't this be `typeof(f)`?

BTW, have you done some benchmarking to check there are no allocations?

Reply:
AFAIK if I wanted to use `typeof{f}` then I would have to store `f` in a struct which would be slower and would allocate more.

I will post benchmarks for the current implementation in the main thread so that they do not disappear.

Reply:
I think in most practical uses, the struct would never allocate. It's the same idea w/ the new iteration protocol: every call to `iterate` potentially returns a `Tuple{T...}`, but most of the time those tuples and even intermediate allocated objects don't even get allocated.